src/Cryptol/Eval/Reference.lhs

```diff
@@ -1752,16 +1763,55 @@ true for all supported precisions.
 >                           y <- fromVFloat <$> yv
 >                           pure (VBit (FP.bfCompare x y == EQ))
 >
->    , "fpIsFinite" ~> vFinPoly \_ -> pure $
->                      vFinPoly \_ -> pure $
+>    , "fpIsNaN"     ~> vFinPoly \_ -> pure $
```

Comment:
What set of operations are we targeting here?

> Implement all floating-point primitives

Are we wanting to support all IEEE-754-2019 operations? We are getting close to that with this set (minus the recommended ones in the standard), or is this driven by another programming language target?

Reply:
By "primitives", I mean all of the Cryptol primitive functions listed in [`lib/Float.cry`](https://github.com/GaloisInc/cryptol/blob/25423ce4ee65337bb75c6922e269a6d359c150e5/lib/Float.cry). (Perhaps unsurprisingly, this list of primitives closely corresponds with the operations provided by SMT-LIB's floating-point theory.)